compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorization.cpp

```diff
@@ -85,7 +83,8 @@ static void populateVectorUnrollPatterns(RewritePatternSet &patterns,
 namespace {
 struct LLVMGPUVectorizationPass
     : public LLVMGPUVectorizationBase<LLVMGPUVectorizationPass> {
-  LLVMGPUVectorizationPass(int64_t nativeVector) : nativeVector(nativeVector) {}
+  LLVMGPUVectorizationPass(int64_t nativeVector, bool generateContract)
```

Comment:
Instead of adding explicit arguments this way, I have found it better to do something like this (https://github.com/llvm/llvm-project/issues/53145) . It is more scalable (you dont need to keep changing all call sites and could be auto-generated in future (if someone gets to that bug). I tend to use this where possible. Maybe a good idea here as well. This also gets a consistent view of command line options as well as when calling pass from within a pipeline.

Reply:
right, so the advantage is that it is controlable from command line? Right now I don't get auto-generated for free?

Reply:
yeah same options from command line or from pipeline. So more consistent in terms of testing. auto-generated would be nice. Maybe ping on that issue :)

Reply:
changed it

Reply:
Not exactly what the bug above says, but its not critical.